internal/operator-controller/authorization/rbac_test.go

```diff
@@ -0,0 +1,305 @@
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: RoleBinding
+metadata:
+  name: test-extension-binding
+  namespace: test-namespace
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: Role
+  name: test-extension-role
+subjects:
+- kind: ServiceAccount
+  name: test-serviceaccount
+  namespace: test-namespace
+  `
+
+	saName = "test-serviceaccount"
+	ns     = "test-namespace"
+
+	objects = []client.Object{
+		&corev1.Namespace{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "test-namespace",
+			},
+		},
+		&rbacv1.ClusterRoleBinding{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "admin-clusterrole-binding",
+			},
+			Subjects: []rbacv1.Subject{
+				{
+					Kind:      "ServiceAccount",
+					Name:      saName,
+					Namespace: ns,
+				},
+			},
+			RoleRef: rbacv1.RoleRef{
+				Name: "admin-clusterrole",
+				Kind: "ClusterRole",
+			},
+		},
+		&corev1.ServiceAccount{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "test-serviceaccount",
+				Namespace: "test-namespace",
+			},
+		},
+	}
+
+	privilegedClusterRole = &rbacv1.ClusterRole{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "admin-clusterrole",
+		},
+		Rules: []rbacv1.PolicyRule{
+			{
+				APIGroups: []string{"*"},
+				Resources: []string{"*"},
+				Verbs:     []string{"*"},
+			},
+		},
+	}
+
+	limitedClusterRole = &rbacv1.ClusterRole{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "admin-clusterrole",
+		},
+		Rules: []rbacv1.PolicyRule{
+			{
+				APIGroups: []string{""},
+				Resources: []string{""},
+				Verbs:     []string{""},
+			},
+		},
+	}
+
+	escalatingClusterRole = &rbacv1.ClusterRole{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "admin-clusterrole",
+		},
+		Rules: []rbacv1.PolicyRule{
+			{
+				APIGroups: []string{"*"},
+				Resources: []string{"serviceaccounts", "services", "certificates"},
+				Verbs:     []string{"*"},
+			},
+			{
+				APIGroups: []string{"rbac.authorization.k8s.io"},
+				Resources: []string{"roles", "clusterroles", "rolebindings", "clusterrolebindings"},
+				Verbs:     []string{"get", "patch", "watch", "list", "create", "update", "delete", "escalate", "bind"},
+			},
+		},
+	}
+)
+
+func setupFakeClient(role client.Object) client.Client {
+	s := runtime.NewScheme()
+	_ = corev1.AddToScheme(s)
+	_ = rbacv1.AddToScheme(s)
+	restMapper := testrestmapper.TestOnlyStaticRESTMapper(s)
+	// restMapper := meta.NewDefaultRESTMapper(nil)
+	fakeClientBuilder := fake.NewClientBuilder().WithObjects(append(objects, role)...).WithRESTMapper(restMapper)
+	return fakeClientBuilder.Build()
+}
+
+func TestPreAuthorize_Success(t *testing.T) {
+	t.Run("preauthorize succeeds with no missing rbac rules", func(t *testing.T) {
+		featuregatetesting.SetFeatureGateDuringTest(t, features.OperatorControllerFeatureGate, features.PreflightPermissions, true)
+		fakeClient := setupFakeClient(privilegedClusterRole)
+		preAuth := NewRBACPreAuthorizer(fakeClient)
+		testServiceAccount := user.DefaultInfo{Name: fmt.Sprintf("system:serviceaccount:%s:%s", ns, saName)}
```

Comment:
we can DRY up these testServiceAccount initializations by making it a global since I think we use the exact same value every time

Reply:
done in e2f3968f